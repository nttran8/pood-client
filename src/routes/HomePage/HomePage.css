.HomePage {
  padding: 25px 25px 20px 25px;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
}

.sec {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.Page__left {
  display: none;
}

#Page__right {
  height: 100%;
}

/* Mobile view and images */
.Page__todo {
  margin: 40px auto 0 auto;
  padding: 3px 5px 3px 5px;
}

.Page__todo > img {
  width: 300px;
  background: white;
  border-radius: 50%;
}

.Page__todo > p {
  font-size: 20px;
  margin-bottom: 20px;
  text-align: left;
  color: white;
}

@media only screen and (min-width: 576px) {
  .HomePage {
    padding: 60px 10px 20px 10px;
    flex-direction: row;
    justify-content: space-around;
  }

  .Page__left {
    display: block;
  }

  .Page__left > p {
    font-size: 23px;
    width: 40vw;
    text-align: left;
  }

  .vision {
    margin-top: 50px;
    width: 40vw;
  }

  #Page__right {
    display: none;
  }

  .Page__right {
    width: 40vw;
    height: 100vh;
    gap: 7rem;
  }
}

@media only screen and (min-width: 768px) {
  .HomePage {
    padding: 50px 20px 20px 20px;
    justify-content: space-evenly;
  }

  .Page__left {
    margin-left: 40px;
  }

  .Page__left > p {
    font-size: 25px;
    font-weight: 600;
    width: 40vw;
  }

  .vision {
    position: relative;
    left: 13%;
    bottom: 18%;
    width: 35vw;
  }

  .Page__right {
    padding: 0 2rem;
    gap: 7rem;
  }
}
