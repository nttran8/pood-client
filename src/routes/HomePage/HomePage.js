// Libraries
import React, { Component } from 'react';

// Service
import TokenService from '../../services/token-service';

// Component
import LoginForm from '../../components/LoginForm/LoginForm';

// Style
import './HomePage.css';
import dooerImg from '../../img/Dooers.svg';


export default class HomePage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {}
    }
  };

  handleLoginRedirect = (userToken) => {
    // Render dashboard if login is successful
    const { location, history } = this.props;
    const destination = (location.state || {}).from || '/dashboard';
    history.push(destination);
  };

  handleRegistrationRedirect = () => {
    // Render homepage/login page
    const { history } = this.props;
    history.push('/register');
  };

  loggedIn = () => {
    // Check if user is logged in
    // if(user.name === 'nt') {
    //   return true;
    // }
    return false;
  };

  render() {
    return (
      <section className="HomePage">
        <div className="Page__left sec">
          <p>Our vision is to inspire and unite a world of doo-ers.</p>
          <img className="vision" src={dooerImg} alt="friends high fiving"/>
        </div>
        <div className="Page__right sec">
          {TokenService.hasAuthToken()
          ? <img alt="poo'd logo"/>
          : <LoginForm onLogin={this.handleLoginRedirect} onRegister={this.handleRegistrationRedirect}/> 
          }
        </div>
      </section>
    );
  }
}