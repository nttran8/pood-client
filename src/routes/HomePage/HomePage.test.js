import React from "react";
import TestRenderer from "react-test-renderer";
import HomePage from "./HomePage";
import { library } from "@fortawesome/fontawesome-svg-core";
import { faUser, faLock } from "@fortawesome/free-solid-svg-icons";
library.add(faUser, faLock);

it(`Renders HomePage UI as expected`, () => {
  const tree = TestRenderer.create(<HomePage />).toJSON();
  expect(tree).toMatchSnapshot();
});
