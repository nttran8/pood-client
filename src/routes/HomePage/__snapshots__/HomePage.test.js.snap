// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders HomePage UI as expected 1`] = `
<section
  className="HomePage"
>
  <div
    className="Page__left sec"
  >
    <p>
      Poo'd is created to inspire and unite a world of doo-ers like you. Let's start logging now!
    </p>
    <img
      alt="friends high fiving"
      className="vision"
      src="Dooers.svg"
    />
  </div>
  <div
    className="Page__right sec"
    id=""
  >
    <div
      className="LoginForm__logoBox"
    >
      <img
        alt="logo"
        className="LoginForm__logo"
        src="Pood.svg"
      />
    </div>
    <form
      className="LoginForm"
      onSubmit={[Function]}
    >
      <div
        className="formBox"
      >
        <h2>
          Login
        </h2>
        <div
          role="alert"
        />
        <div
          className="inputBox"
        >
          <label
            aria-label="username"
            htmlFor="username"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-user fa-w-14 "
              data-icon="user"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </label>
          <input
            className="input username publicPlaceholder"
            id="username"
            placeholder="Username"
          />
        </div>
        <div
          className="inputBox"
        >
          <label
            aria-label="password"
            htmlFor="password"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-lock fa-w-14 "
              data-icon="lock"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </label>
          <input
            className="input password publicPlaceholder"
            id="password"
            placeholder="Password"
            type="password"
          />
        </div>
        <button
          className="button "
          type="submit"
        >
          Sign in
        </button>
        <button
          className="button "
          onClick={[Function]}
          type="button"
        >
          Register
        </button>
      </div>
    </form>
  </div>
</section>
`;
